diff --git a/ldso/include/ldso.h b/ldso/include/ldso.h
index 6f3b728..e72d9da 100644
--- a/ldso/include/ldso.h
+++ b/ldso/include/ldso.h
@@ -48,6 +48,7 @@
 #ifdef __UCLIBC_HAS_TLS__
 /* Defines USE_TLS */
 #include <tls.h>
+#include <dl-tls.h>
 #endif
 #include <dl-hash.h>
 
diff --git a/libpthread/nptl/sysdeps/mips/dl-tls.h b/libpthread/nptl/sysdeps/mips/dl-tls.h
index 3ad3b2b..fbc2c8e 100644
--- a/libpthread/nptl/sysdeps/mips/dl-tls.h
+++ b/libpthread/nptl/sysdeps/mips/dl-tls.h
@@ -17,6 +17,8 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+#ifndef _DL_TLS_H
+#define _DL_TLS_H	1
 
 /* Type used for the representation of TLS information in the GOT.  */
 typedef struct
@@ -44,3 +46,5 @@ extern void *__tls_get_addr (tls_index *ti);
 
 # define GET_ADDR_OFFSET	(ti->ti_offset + TLS_DTV_OFFSET)
 # define __TLS_GET_ADDR(__ti)	(__tls_get_addr (__ti) - TLS_DTV_OFFSET)
+
+#endif	/* dl-tls.h */
